src/actions/sso/credentialOfferRequest.ts

```diff
@@ -62,23 +94,86 @@ export const consumeCredentialOfferRequest = (
     CredentialsReceive
   >(res.token)
 
-  return dispatch(
-    withLoading(
-      withErrorScreen(
-        receiveExternalCredential(
-          credentialReceive,
-          requesterSummary,
-          isDeepLinkInteraction,
-          selectedMetadata,
-        ),
-        err => new AppError(ErrorCode.CredentialsReceiveFailed, err),
-      ),
-    ),
+  await identityWallet.validateJWT(credentialReceive, undefined, registry)
+
+  const providedCredentials =
+    credentialReceive.interactionToken.signedCredentials
+
+  if (!providedCredentials.length) throw new Error('No credentials received')
+
+  const validationResults = await JolocomLib.util.validateDigestables(
+    providedCredentials,
+  )
+
+  // TODO Error Code
+  if (validationResults.includes(false)) {
+    throw new Error('Invalid credentials received')
+  }
+
+  providedCredentials.map(async credential => {
+    await storageLib.delete.verifiableCredential(credential.id)
+    await storageLib.store.verifiableCredential(credential)
+  })
+
+  const issuerDid = keyIdToDid(credentialOfferRequest.issuer)
+  const offerCredentialDetails = assembleCredentialDetails(
+    interactionToken,
+    issuerDid,
   )
+  if (offerCredentialDetails) {
```

Comment:
if `offerCredentialDetails` is deduplicated first, we can use `Promise.all()` and let the calls run in parallel. Ideally we should batch write them to the DB.